lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -908,6 +923,33 @@ public SimplePage getPage(long pageId) {
 		}
 	}
 
+	public String getPageUrl(long pageId) {
+
+		List<SimplePageItem> pageItems = findPageItemsBySakaiId(Long.toString(pageId));
+		if (pageItems.size() == 0) {
+			log.error("No page items found for lessons page with id: {}", pageId);
+			return null;
+		}
+		long pageItemId = pageItems.get(0).getId();
+		SimplePage page = getPage(pageId);
+		SitePage sitePage = siteService.findPage(page.getToolId());
+		if (sitePage == null) {
+			log.error("Failed to find Sakai SitePage for lessons page with id: {}", pageId);
+		}
+		String siteId = sitePage.getSiteId();
```

Comment:
Won't this just NPE if there isn't a sitePage, why log 2 lines earlier and then let it blow up as well?